otelcol/configprovider.go

```diff
@@ -105,26 +97,7 @@ func (cm *configProvider) Get(ctx context.Context, factories Factories) (*Config
 
 	var cfg *configSettings
 	if cfg, err = unmarshal(conf, factories); err != nil {
-		err = fmt.Errorf("cannot unmarshal the configuration: %w", err)
-
-		if globalgates.StrictlyTypedInputGate.IsEnabled() {
-			var shouldAddCoda bool
```

Comment:
I'm likely missing context here, but in other places in this PR we've kept the behaviour inside the check if the `StrictlyTypedInputGate.IsEnabled` and here it's removed.

Reply:
The coda says:

```
Hint: Temporarily restore the previous behavior by disabling 
      the `confmap.strictlyTypedInput` feature gate. More details at:
      https://github.com/open-telemetry/opentelemetry-collector/issues/10552
```

which is not actionable after this PR is merged (you cannot disable the feature gate). Should we keep some sort of custom message? It could make sense to give people a place where they can complain